File: core-presentation/src/main/java/home/oleg/placesnearme/core_presentation/utils/DistanceUtil.java
package home.oleg.placesnearme.core_presentation.utils;

import android.content.Context;

import home.oleg.placesnearme.core_presentation.R;

/**
 * Created by Oleg Sheliakin on 23.10.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
public final class DistanceUtil {

    public static final double KILOMETER = 1000.0;

    private DistanceUtil() {
    }

    public static String convertDistanceTOString(double distance, Context context) {
        if (distance >= 1000) {
            double tempKm = distance / KILOMETER;
            return context.getString(R.string.kms, tempKm);
        } else {
            return context.getString(R.string.meters, (int) distance);
        }
    }

}

File: core-presentation/src/main/java/home/oleg/placesnearme/corepresentation/utils/DistanceUtil.kt
package home.oleg.placesnearme.corepresentation.utils

import android.content.Context

import home.oleg.placesnearme.corepresentation.R

/**
 * Created by Oleg Sheliakin on 23.10.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
object DistanceUtil {

    private const val KILOMETER = 1000.0

    fun convertDistanceToString(distance: Double, context: Context): String {
        if (distance >= 1000) {
            val tempKm = distance / KILOMETER
            return context.getString(R.string.kms, tempKm)
        } else {
            return context.getString(R.string.meters, distance.toInt())
        }
    }

}
